lbplanner/services/user/update_user.php

```diff
@@ -16,88 +16,101 @@
 
 namespace local_lbplanner_services;
 
+use dml_exception;
 use external_api;
 use external_function_parameters;
 use external_single_structure;
 use external_value;
+use invalid_parameter_exception;
 use local_lbplanner\helpers\user_helper;
 use local_lbplanner\helpers\plan_helper;
+use moodle_exception;
 
 /**
  * Update the data for a user.
  */
 class user_update_user extends external_api {
-    public static function update_user_parameters() {
+    public static function update_user_parameters(): external_function_parameters {
         return new external_function_parameters(array(
-            'userid' => new external_value(PARAM_INT, 'The id of the user to register', VALUE_REQUIRED, null, NULL_NOT_ALLOWED),
-            'lang' => new external_value(PARAM_TEXT, 'The language the user has selected', VALUE_REQUIRED, null, NULL_NOT_ALLOWED),
-            'theme' => new external_value(PARAM_TEXT, 'The theme the user has selected', VALUE_REQUIRED, null, NULL_NOT_ALLOWED),
+            'lang' => new external_value(PARAM_TEXT, 'The language the user has selected', VALUE_DEFAULT, null),
+            'theme' => new external_value(PARAM_TEXT, 'The theme the user has selected', VALUE_DEFAULT, null),
             'colorblindness' => new external_value(
                 PARAM_TEXT,
                 'The colorblindness the user has selected',
-                VALUE_REQUIRED,
-                null,
-                NULL_NOT_ALLOWED),
+                VALUE_DEFAULT,
+                null),
             'displaytaskcount' => new external_value(
                 PARAM_INT,
                 'The displaytaskcount the user has selected',
```

Comment:
same here this is not very telling what 'displaytaskcount' is. maybe also add that the int must be either `0` (false) or `1` (true) ?